### PR TITLE
[openthread] update to commit 5121754

### PR DESCRIPTION
**Pull request by huamenggg**
This commit update OpenThread code to the latest version.

### CODE REVIEW

File: src/agent/ncp_openthread.cpp [hunk 1]
```diff
@@ -192,4 +195,10 @@ extern "C" void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const ch
 } // namespace BorderRouter
 } // namespace ot
 
+void otPlatReset(otInstance *aInstance)
```
Comment (bukepo) [inline src/agent/ncp_openthread.cpp]: This must be properly implemented in order to pass the travis-ci tests.

File: src/agent/agent_instance.hpp [hunk 2]
```diff
@@ -89,6 +89,14 @@ class AgentInstance
      */
     void Process(const otSysMainloopContext &aMainloop);
 
+    /**
+     * This method return mNcp pointer.
+     *
+     * @retval  the pointer of mNcp.
+     *
+     */
+    Ncp::Controller *getNcp() { return mNcp; }
```
Comment (jwhui) [inline src/agent/agent_instance.hpp]: style: All void functions or methods shall explicitly declare and specify the void type keyword.

File: src/agent/agent_instance.hpp [hunk 3]
```diff
@@ -95,7 +95,7 @@ class AgentInstance
      * @retval  the pointer of mNcp.
      *
      */
-    Ncp::Controller *getNcp(void) { return mNcp; }
+    Ncp::Controller *GetNcp(void) { return mNcp; }
```
Comment (bukepo) [inline src/agent/agent_instance.hpp]: better to return reference?

File: src/agent/main.cpp [hunk 4]
```diff
@@ -77,7 +75,9 @@ static int Mainloop(AgentInstance &aInstance)
 
     while (true)
     {
-        otSysMainloopContext mainloop;
+        otSysMainloopContext                         mainloop;
+        ot::BorderRouter::Ncp::ControllerOpenThread *ncp =
```
Comment (bukepo) [inline src/agent/main.cpp]: we are using namespace `ot::BorderRouter`, just `Ncp::Controller`. do we need use concrete `ControllerOpenThread`?

File: src/agent/main.cpp [hunk 5]
```diff
@@ -88,32 +88,29 @@ static int Mainloop(AgentInstance &aInstance)
 
         aInstance.UpdateFdSet(mainloop);
 
-        if (select(mainloop.mMaxFd + 1, &mainloop.mReadFdSet, &mainloop.mWriteFdSet, &mainloop.mErrorFdSet,
-                   &mainloop.mTimeout) >= 0)
+        int rval = select(mainloop.mMaxFd + 1, &mainloop.mReadFdSet, &mainloop.mWriteFdSet, &mainloop.mErrorFdSet,
```
Comment (bukepo) [inline src/agent/main.cpp]: declare under line 78?

File: src/agent/ncp_openthread.hpp [hunk 6]
```diff
@@ -75,7 +75,23 @@ class ControllerOpenThread : public Controller
...
+     */
+    virtual void SetReset(bool status) { mIsNcpControllerReset = status; }
+
+    /**
+     * This method get mIsNcpControllerReset.
+     *
+     * @retval  the bool value of mIsNcpControllerReset.
+     *
+     */
+    virtual bool GetIsReset(void) { return mIsNcpControllerReset; }
```
Comment (bukepo) [inline src/agent/ncp_openthread.hpp]: -> `IsResetting`?

**Comment by bukepo** [comment]
@huamenggg I think the latest OpenThread can get this PR passed.

**Pull request merged**